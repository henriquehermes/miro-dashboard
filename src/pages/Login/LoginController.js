import React, { useState } from 'react';
import PropTypes from 'prop-types';

import LoginComponent from './LoginComponent';

const LoginController = ({ history }) => {
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');

  async function handleSubmit(event) {
    event.preventDefault();

    if (username === 'admin' && password === 'admin') {
      localStorage.setItem('username', username);
      localStorage.setItem('admin', true);
      history.push('/dashboard');
    } else if (username === 'user' && password === 'user') {
      localStorage.setItem('username', username);
      history.push('/dashboard');
    } else {
      // eslint-disable-next-line no-alert
      alert('User not found.');
    }
  }

  return (
    <LoginComponent
      handleSubmit={e => handleSubmit(e)}
      username={username}
      setUsername={setUsername}
      password={password}
      setPassword={setPassword}
    />
  );
};

LoginController.propTypes = {
  history: PropTypes.func.isRequired,
};

export default LoginController;
