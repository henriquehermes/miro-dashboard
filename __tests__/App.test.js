import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import LoginController from '../src/pages/Login/LoginController';

const theme = {
  primary: '#00bb77',
  secondary: '#282a36',
};

describe('Login', () => {
  test('Render login', () => {
    const wrapper = renderer.create(<LoginController theme={theme} />).toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
