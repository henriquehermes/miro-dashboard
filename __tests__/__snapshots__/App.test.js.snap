// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Render login 1`] = `
.c4 {
  height: 40px;
  width: 100%;
  margin: 10px 0;
  padding: 0 10px;
  border: none;
  background: #fff;
  border-radius: 6px;
  border-color: #fff;
  border-style: solid;
  box-shadow: 0px 4px 18px 0px rgba(0,0,0,0.1);
  font-weight: bold;
}

.c6 {
  cursor: pointer;
  height: 40px;
  width: 90px;
  font-weight: bold;
  background: #fff;
  border-radius: 6px;
  border-color: #fff;
  border-style: solid;
  box-shadow: 0px 4px 18px 0px rgba(0,0,0,0.1);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  height: 100vh;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  background: #f6f6f6;
  padding: 20px 16px;
  border-radius: 6px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 350px;
  width: 450px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  font-size: 30px;
  font-weight: bold;
  margin-bottom: 30px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 80%;
}

.c3 > label {
  font-weight: bold;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100%;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <h1
      className="c2"
    >
      Bem Vindo
    </h1>
    <form
      className="c3"
    >
      <label>
        Username
      </label>
      <input
        className="c4"
        id="username"
        onChange={[Function]}
        type="text"
        value=""
      />
      <label>
        Password
      </label>
      <input
        className="c4"
        id="username"
        onChange={[Function]}
        type="password"
        value=""
      />
      <div
        className="c5"
      >
        <button
          className="c6"
          onClick={[Function]}
        >
          Login
        </button>
      </div>
    </form>
  </div>
</div>
`;
